### PR TITLE
Throw exception on transactions in eventloop threads

### PR DESCRIPTION
**Pull request by philippguertler**

### CODE REVIEW

File: core/src/main/java/com/gentics/mesh/core/data/job/impl/NodeMigrationJobImpl.java [hunk 1]
```diff
@@ -60,10 +61,10 @@ private SchemaMigrationMeshEventModel createEvent(MeshEvent event, JobStatus sta
 		return model;
 	}
 
-	private NodeMigrationActionContextImpl prepareContext() {
```
Comment (Jotschi) [inline core/src/main/java/com/gentics/mesh/core/data/job/impl/NodeMigrationJobImpl.java]: Ensure that MicronodeMigration and BranchMigration all work the same way. Either return Single or the context.

File: core/src/main/java/com/gentics/mesh/core/verticle/job/JobWorkerVerticle.java [hunk 2]
```diff
@@ -54,10 +54,10 @@ public String getLockName() {
 
 	@Override
 	public Completable executeJob(Message<Object> message) {
-		return Completable.defer(() -> db.tx(() -> {
+		return db.singleTx(() -> {
 			JobRoot jobRoot = boot.get().jobRoot();
 			return jobRoot.process();
-		}));
+		}).flatMapCompletable(x -> x);
```
Comment (Jotschi) [inline core/src/main/java/com/gentics/mesh/core/verticle/job/JobWorkerVerticle.java]: Use identity helper here as well

File: elasticsearch/src/main/java/com/gentics/mesh/search/index/AbstractSearchHandler.java [hunk 3]
```diff
@@ -212,7 +217,7 @@ public void rawQuery(InternalActionContext ac) {
 
 		RL listResponse = classOfRL.newInstance();
 
-		waitUtil.awaitSync(ac).andThen(Single.defer(() -> {
+		waitUtil.awaitSync(ac).andThen(executeBlocking(vertx, () -> {
```
Comment (Jotschi) [inline elasticsearch/src/main/java/com/gentics/mesh/search/index/AbstractSearchHandler.java]: Why was this not changed to singleTx?

File: elasticsearch/src/main/java/com/gentics/mesh/search/verticle/eventhandler/GroupUserAssignmentHandler.java [hunk 4]
```diff
@@ -36,13 +36,11 @@ public GroupUserAssignmentHandler(MeshHelper helper, MeshEntities entities) {
...
 	public Flowable<? extends SearchRequest> handle(MessageEvent messageEvent) {
-		return Flowable.defer(() -> {
-			GroupUserAssignModel model = requireType(GroupUserAssignModel.class, messageEvent.message);
-			return Flowable.just(helper.getDb().tx(() -> {
-				User user = helper.getBoot().userRoot().findByUuid(model.getUser().getUuid());
-				return entities.createRequest(user);
-			}));
-		});
+		GroupUserAssignModel model = requireType(GroupUserAssignModel.class, messageEvent.message);
+		return helper.getDb().singleTx(() -> {
```
Comment (Jotschi) [inline elasticsearch/src/main/java/com/gentics/mesh/search/verticle/eventhandler/GroupUserAssignmentHandler.java]: Should this be directly run in a tx ?

File: elasticsearch/src/main/java/com/gentics/mesh/search/verticle/eventhandler/SimpleEventHandler.java [hunk 5]
```diff
@@ -51,11 +53,11 @@ public SimpleEventHandler(MeshHelper helper, MeshEntity<T> entity, String indexN
 			MeshElementEventModel model = requireType(MeshElementEventModel.class, eventModel.message);
 
 			if (event == entity.getCreateEvent() || event == entity.getUpdateEvent()) {
-				return toFlowable(helper.getDb().tx(() -> entity.getDocument(model))
+				return toFlowable(helper.getDb().singleTxImmediate(() -> entity.getDocument(model)))
 					.map(document -> helper.createDocumentRequest(
 						indexName, model.getUuid(),
 						document, complianceMode
-					)));
+					));
```
Comment (Jotschi) [inline elasticsearch/src/main/java/com/gentics/mesh/search/verticle/eventhandler/SimpleEventHandler.java]: Change to `.toFlowable()` if possible.

File: elasticsearch/src/main/java/com/gentics/mesh/search/verticle/eventhandler/project/ProjectUpdateEventHandler.java [hunk 6]
```diff
@@ -101,16 +102,15 @@ public ProjectUpdateEventHandler(MeshHelper helper, MeshEntities entities, MeshO
...
 				.flatMap(project -> project.getTagFamilyRoot().findAll().stream()
 				.flatMap(family -> Stream.concat(
 					Stream.of(createTagFamilyRequest(project, family)),
 					createTagRequests(family, project)
 				)
 			))
 			.collect(toFlowable()))
-			.runInNewTx()
-		);
+			.runInAsyncTx().flatMapPublisher(RxUtil.identity());
```
Comment (Jotschi) [inline elasticsearch/src/main/java/com/gentics/mesh/search/verticle/eventhandler/project/ProjectUpdateEventHandler.java]: Add linebreak

File: elasticsearch/src/main/java/com/gentics/mesh/search/index/AbstractSearchHandler.java (hunk 3 above)
Comment (philippguertler) [reply elasticsearch/src/main/java/com/gentics/mesh/search/index/AbstractSearchHandler.java]: getSelectedIndices in this lambda already opens a transaction. This lambda does not require a transaction itself, so executeBlocking is used instead.

Comment (philippguertler) [reply elasticsearch/src/main/java/com/gentics/mesh/search/index/AbstractSearchHandler.java]: The above rawQuery was now changed to executeBlocking as well.

**Pull request closed without merging**